File: gb-spring-backend/src/main/java/com/egroupbuy/model/Order.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Order {
    int order_id;
    Timestamp order_date;
    double sub_total;
    double shipping_fee;
    double tax;
    double order_total;
    String shipping_firstname;
    String shipping_lastname;
    String shipping_email;
    String shipping_phone;
    String shipping_address;
    String shipping_country;
    String shipping_post_code;
    String payment_status;
    String payment_method;
    String progress_status;
    int buyer_id;
    String buyer_username;
    String buyer_email;
    String buyer_firstname;
    String buyer_lastname;
    String buyer_phone;
    String buyer_address;
    String buyer_country;
    String buyer_post_code;
    int seller_id;
    String seller_username;
    int product_id;
    String product_title;
    int package_id;
    String package_name;
    int quantity;
    String shipping_method;
    double gateway_fee;
    String card_holder_name;
    String card_number;
    String card_expiry_date;
    String card_cvv;
    List<OrderDetail> order_detail_list;
}

File: gb-angular-frontend/src/app/interfaces/order.ts
import {OrderDetail} from './order-detail';

export interface Order {
  order_id?: number;
  order_date: string;
  sub_total: number;
  shipping_fee: number;
  tax: number;
  order_total: number;

  shipping_firstname: string;
  shipping_lastname: string;
  shipping_email: string;
  shipping_phone: string;

  shipping_address: string;
  shipping_country: string;
  shipping_post_code: string;
  payment_status: string;
  payment_method: string;
  progress_status: string;
  buyer_id: number;
  buyer_username: string;
  buyer_email:string;

  buyer_firstname: string;
  buyer_lastname: string;
  buyer_phone: string;
  buyer_address: string;
  buyer_country: string;
  buyer_post_code: string;

  seller_id: number;
  seller_username: string;
  product_id: number;
  product_title: string;
  package_id: number;
  package_name: string
  quantity: number;

  shipping_method: string;
  gateway_fee: number;
  card_holder_name: string;
  card_number: string;
  card_expiry_date: string;
  card_cvv: string;
  order_detail_list: OrderDetail[];
}
